### ISSUE TITLE
Creating big fixed matrices?

### ISSUE THREAD

**Issue by tueda** [bluss/ndarray#446]
I was trying to test and benchmark a routine whose input has an integer matrix. I created the matrix as
```rust
let a: Array2<u32> = arr2(&[
  ...
]);
```
but when I tried to put a 32 by 32 matrix, it failed:
```
158 |     let a: Array2<u32> = arr2(&[
    |                          ^^^^ the trait `ndarray::FixedInitializer` is not implemented for `[{integer}; 32]`
```
Using the `array!` macro gave me a similar error.

So I was wondering about the maximum number of elements for `FixedInitializer`:
https://github.com/bluss/ndarray/blob/98d20ef59f3d089630978144243bb0b52e92dded/src/free_functions.rs#L184
If this is extended to `32`, my example would work. But then one may argue how about a larger number like `64`. It would be very nice if the limitation of the maximum size could be removed away somehow, but I'm not sure if it could be possible in the Rust language.

Or, in general, do you have some recommended way to create such a big (both in rows and columns) constant matrix?

**Comment by jturner314** [comment]
`FixedInitializer` is limited to fixed sizes because Rust currently doesn't support operating in a generic way over constants. Once rust-lang/rust#44580 is resolved, we can remove this limitation.

For the time being, here are a few options to construct a large matrix:

1. Use one of the various [constructor methods](https://docs.rs/ndarray/0.11.2/ndarray/struct.ArrayBase.html#constructor-methods-for-owned-arrays) to generate the array. For example, if your array consists of a single element value, you could use `::from_elem()`, or if your array values are a function of the indices, you could use `::from_shape_fn()`.
2. Create an array of [zeros](https://docs.rs/ndarray/0.11.2/ndarray/struct.ArrayBase.html#method.zeros) or an [uninitialized](https://docs.rs/ndarray/0.11.2/ndarray/struct.ArrayBase.html#method.uninitialized) array and then assign the values.
3. Use the `::from_shape_vec()` method on the result of the `vec![]` macro. For example, you could do `Array::from_shape_vec((2, 2), vec![1, 2, 3, 4])`. (For your case, change the shape from `(2, 2)` to `(32, 32)` and add the remaining element values.)
4. If all you need is a view and not an owned array, then you could use `ArrayView::from_shape()`. For example, you could do `ArrayView::from_shape((2, 2), &[1, 2, 3, 4])`.

In summary, you can use one of the existing constructor methods.

Does that help?

**Comment by tueda** [comment]
Thank you for your answer and explaining the background of the implementation. Actually now I find what I need is a view `ArrayView2` for a constant matrix (each element is chosen quasi-randomly by another program), and so the answer 4 works well!
